includes/gateways/functions.php

```diff
@@ -67,12 +72,18 @@ function give_get_payment_gateways($version = 2)
  * @param int|null $version
  * @return array $gateway_list All the available gateways
  */
-function give_get_enabled_payment_gateways($form_id = 0, $version = 2)
+function give_get_enabled_payment_gateways($form_id = 0, $version = null)
 {
-    $suffix = $version === 3 ? '_v3' : '';
     $gateways = give_get_payment_gateways($version);
 
-    $enabled = $_POST['gateways' . $suffix] ?? give_get_option('gateways' . $suffix);
+    $enabled = [];
+    if (!$version || $version === 2) {
+        $enabled = array_merge($enabled, $_POST['gateways'] ?? (array)give_get_option('gateways', []));
+    }
+
+    if (!$version || $version === 3) {
+        $enabled = array_merge($enabled, $_POST['gateways_v3'] ?? (array)give_get_option('gateways_v3', []));
+    }
```

Comment:
Since we're using `array_merge` now, let's make sure we're validating / casting the `$_POST` inputs to an array to prevent a potential type error.

Very verbose example 😄 

```
$gatewaysFromPostRequest = isset($_POST['gateways']) && is_array($_POST['gateways']) ? (array)$_POST['gateways'] : null;
```

Reply:
We don't need that level of verbosity 😆 
If it's already an array, there's no reason to cast it. So casting any value does do the trick for us.

08fbab2fcdd384ae5c6550d00056edc5ee3ef7dc